pkg/utils/k8sutil.go

```diff
@@ -819,8 +819,40 @@ func EnsureFuncDeployment(client kubernetes.Interface, funcObj *spec.Function, o
 	return err
 }
 
+func doRESTReq(restIface rest.Interface, groupVersion, verb, resource, elem, namespace string, body *[]byte) ([]byte, error) {
+	var req *rest.Request
+	switch verb {
+	case "get":
+		req = restIface.Get().Name(elem)
+		break
+	case "create":
+		req = restIface.Post().Body(*body)
+		break
+	case "update":
+		req = restIface.Put().Name(elem).Body(*body)
+		break
+	default:
+		return []byte{}, fmt.Errorf("Verb %s not supported", verb)
+	}
+	return req.AbsPath("apis", groupVersion, "namespaces", namespace, resource).DoRaw()
+}
+
+func doRESTCronJobReq(restIface rest.Interface, groupVersion, verb, elem, namespace string, body *[]byte) (batchv2alpha1.CronJob, error) {
```

Comment:
You can make this func more generic by not sticking to the returned cronjob object. It can be whatever object (return []byte) and you can do the unmarshaling on that. Then we can call it for other objects in future.

Reply:
my idea was to have the `doRESTReq` as the generic function (it is the one that returns `[]byte`). For this case it is just a matter of convenience so I don't have to do the Unmarshalling each time a do a request.

Reply:
but it is true that we can set `res` as an `interface{}` and we can do the `Unmarshal` in the `doRESTReq`